src/lib/routes/admin-api/project/features.ts

```diff
@@ -109,13 +115,46 @@ export default class ProjectFeaturesController extends Controller {
             DELETE_FEATURE_STRATEGY,
         );
 
-        // feature toggles
-        this.get(PATH, this.getFeatures);
-        this.post(PATH, this.createFeature, CREATE_FEATURE);
+        this.route({
+            method: 'get',
+            path: PATH,
+            handler: this.getFeatures,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    responses: { 200: featuresResponse },
+                }),
+            ],
+        });
+
+        this.route({
+            method: 'post',
+            path: PATH,
+            handler: this.createFeature,
+            permission: CREATE_FEATURE,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    requestBody: createFeatureRequest,
+                    responses: { 200: featureResponse },
+                }),
+            ],
+        });
 
         this.post(PATH_FEATURE_CLONE, this.cloneFeature, CREATE_FEATURE);
 
-        this.get(PATH_FEATURE, this.getFeature);
+        this.route({
```

Comment:
Could we utilize the Controller to allow us to do
```
this.get(PATH_FEATURE, this.getFeature, { tags: ['admin'], responses: { 200: featureResponse }})
```
?

Reply:
I still think this is a lot better than maintaining a yaml file in a completely different location, so if it becomes to cumbersome, I'm defintely up for the route function for documenting openapi :)

Reply:
True, that would work! I mostly kept `OpenApiService` outside `Controller` to avoid having to change every file that `extends Controller`. Maybe a change for later?